HpToolsLauncher/ParallelRunner/ParallelRunnerEnvironmentUtil.cs

```diff
@@ -426,9 +430,29 @@ public static string GetConfigFilePath(TestInfo testInfo, McConnectionInfo mcCon
 
             JavaScriptSerializer serializer = new JavaScriptSerializer();
 
-            var configJson = serializer.Serialize(config);
+            string configJson = null;
 
-            File.WriteAllText(configFilePath, configJson);
+            try
+            {
+                configJson = serializer.Serialize(config);
+            }
+            catch(InvalidOperationException e)
+            {
+                throw new ParallelRunnerConfigurationException("Invalid json confguration provided: ",e);
+            }
+            catch(ArgumentException e)
+            {
+                throw new ParallelRunnerConfigurationException("Configuration serialization recursion limit exceeded: ", e);
+            }
+
+            try
+            {
+                File.WriteAllText(configFilePath, configJson);
+            }
+            catch (Exception e)
```

Comment:
Should we classify each exception with other messages for better error handling?

Reply:
The inner exception is specified when throwing the ParallelRunnerConfigurationException.